### PR TITLE
fixtures: save env information as xml properties

### PR DESCRIPTION
**Pull request by Emantor**
This commit adds the ability to save environment information in the junit xml
file.

I'm not sure we want to do this in the env fixture, but the configuration gets not parsed beforehand and all the necessary information is available here.

TODO:
- [x] Documentation

### CODE REVIEW

File: labgrid/pytestplugin/fixtures.py [hunk 1]
```diff
@@ -52,6 +54,37 @@ def env(request):
...
+            try:
+                name = env.config.get_targets()[j]['resources']['RemotePlace']['name']
+                my_junit.add_global_property('TARGET_{}_REMOTE'.format(i), name)
+            except KeyError:
+                pass
+            my_junit.add_global_property('TARGET_{}'.format(i), j)
+
+        for name,path in env.config.get_paths().items():
+            try:
+                sha = subprocess.check_output("git show --format=%H -s".split(" "),cwd=path)
```
Comment (jluebbe) [inline labgrid/pytestplugin/fixtures.py]: Please use "git rev-parse HEAD" instead.

File: labgrid/pytestplugin/fixtures.py [hunk 2]
```diff
@@ -52,6 +54,37 @@ def env(request):
...
+            except subprocess.CalledProcessError:
+                continue
+            except FileNotFoundError:
+                continue
+            my_junit.add_global_property('PATH_{}'.format(name.upper()),sha.decode("utf-8").strip("\n"))
```
Comment (jluebbe) [inline labgrid/pytestplugin/fixtures.py]: We should always export the path itself. The commit could be exported under PATH_name_GIT_COMMIT.

**Comment by Emantor** [comment]
Changes done, should be ready to merge

File: doc/usage.rst [hunk 3]
```diff
@@ -297,6 +297,20 @@ They can also be converted to other formats, such as HTML with `junit2html tool
   $ pytest --lg-env shell-example.yaml --junit-xml=report.xml
   $ junit2html report.xml
 
+
+Labgrid adds additional xml properties to a test run, these are:
+
+- YAML_CONFIG: Name of the configuration file
```
Comment (jluebbe) [inline doc/usage.rst]: Use ENV_CONFIG instead of YAML_CONFIG (also below).

File: labgrid/pytestplugin/fixtures.py [hunk 4]
```diff
@@ -43,15 +41,55 @@ def env(request):
...
         pytest.skip("missing environment config (use --lg-env)")
     env = Environment(config_file=lg_env)
     if lg_coordinator is not None:
         env.config.set_option('crossbar_url', lg_coordinator)
+
+    if pytest.config.pluginmanager.hasplugin('junitxml'):
+        my_junit = getattr(pytest.config, '_xml', None)
+
+        my_junit.add_global_property('YAML_CONFIG', env.config_file)
+        for i, j in enumerate(env.config.get_targets()):
```
Comment (jluebbe) [inline labgrid/pytestplugin/fixtures.py]: Simplify this.

File: doc/usage.rst [hunk 5]
```diff
@@ -297,6 +297,20 @@ They can also be converted to other formats, such as HTML with `junit2html tool
...
+- TARGET_{NUMBER}: Name of the target
```
Comment (jluebbe) [inline doc/usage.rst]: Make TARGETS a list of target names.

**Comment by Emantor** [comment]
Adressed the changes, should be ready to go.

File: labgrid/config.py [hunk 6]
```diff
@@ -128,3 +128,19 @@ def set_option(self, name, value):
...
+        """
+        return self.data.get('paths', {})
+
+    def get_images(self):
+        """Helper function that returns the subdict of all images
+
+        Returns:
+            Dict: Dictionary containing all image definitions
+        """
+        return self.data.get('images', {})
```
Comment (jluebbe) [inline labgrid/config.py]: Paths and images should also be resolved.

**Comment by Emantor** [comment]
- [X] Resolve paths
- [X] Rebased
=> ready to go 👍

**Pull request merged**